drift.py

```diff
@@ -25,6 +25,7 @@ def __init__(self, specs=None):
 
     def analyze(self, commit):
         for abstract_spec in self.specs:
+            tags = []
```

Comment:
Why do we need to define tags up here if we re-define the variable below?

Reply:
If we define tags within an if statement doesn't that mean the lifetime of the variable is only within the if/else statement and not available outside of that unless it's been previously declared? Maybe Python is different but that's how Go, C, and Java do it.

Reply:
Yes but in Python you don't need to declare anything. Below you have both:

```
                    tags = ["%s:%s" %(x[0],x[1]) for x in diff['b_not_a']]
```
and
```python
                  tags = ["concretization-failed"]
```
so you are defining it outside the loop, and then inside twice. Unless you are updating this list (which you are not) it's not clear to me why you need it there.

Reply:
Wow okay that breaks my brain but you're totally right. Python is weird!